microservices/account-operations/src/main/java/edu/gacademy/accountoperations/clients/AuthenticationClient.java

```diff
@@ -34,12 +34,17 @@ public int getUserId(String token) throws StatusRuntimeException {
     }
 
     private StatusRuntimeException wrapException(StatusRuntimeException e) {
+        if (e.getStatus() != Status.UNKNOWN) {
+            return e;
+        }
         String[] splitMessage = e.getMessage().split(" ", 3);
-        int statusCode;
+        int statusCode = Status.UNKNOWN.getCode().value();
         if (splitMessage.length >= 2) {
```

Comment:
can you explain me this fragment of the code

Reply:
The problem is that grpc for node has extremely bad api for handling errors. That's why I send all of the error information in error message and after recieving it the client parse it to appropriate object